altair/vegalite/v5/schema/_config.py

```diff
@@ -0,0 +1,1055 @@
+# The contents of this file are automatically written by
+# tools/generate_schema_wrapper.py. Do not modify directly.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Literal, Sequence, TypedDict
+
+# ruff: noqa: F405
+if TYPE_CHECKING:
+    from ._typing import *  # noqa: F403
+    from .core import *  # noqa: F403
+    from .core import Dict
+
+
+class RectConfigKwds(TypedDict, total=False):
+    """Placeholder doc."""
+
+    align: Align_T
+    angle: float
+    aria: bool
+    ariaRole: str
+    ariaRoleDescription: str
+    aspect: bool
+    baseline: str | Baseline_T
+    binSpacing: float
+    blend: Blend_T
+    color: str | LinearGradient | RadialGradient | ColorName_T
```

Comment:
TODO: Reduce `LinearGradient | RadialGradient`

Reply:
Also noting here anywhere there is:

```py
color: str | ColorName_T
```

There is potentially a use-case for `typing.Annotated` (mentioned in https://github.com/vega/altair/discussions/3476#discussioncomment-10162300).
The `str` case is for `HexColor`, which should be a simple enough regex pattern to reproduce.

Reply:
> There is potentially a use-case for `typing.Annotated` (mentioned in [#3476 (comment)](https://github.com/vega/altair/discussions/3476#discussioncomment-10162300)). The `str` case is for `HexColor`, which should be a simple enough regex pattern to reproduce.

Looked into further and a possible issue will be using `Annotated` within a `TypedDict`, since there won't be an object available at runtime to introspect.

So we could annotate it here, with a useful name, but utilising the metadata provided (the regex pattern to be matched) would have to happen in `ThemeRegistry` on `.register(theme)`